File: Advanced Methods of Programming/Toy Language Interpreter/src/Model/Statements/Print.java
package Model.Statements;

import Model.Expressions.IExpression;
import Model.ProgramState;

public class Print implements IStatement {
    private IExpression exp;

    public Print(IExpression exp){
        this.exp = exp;
    }

    @Override
    public String toString() {
        return "print(" + exp.toString() + ")";
    }

    public ProgramState exec(ProgramState state){
        state.appendOutput(exp.eval(state) + "\n");
        return state;
    }
}

File: Advanced Methods of Programming/CSharp Toy Language Interpreter/ToyLanguageInterpreter/Src/Model/Statements/Print.cs
using System;
using ToyLanguageInterpreter.Src.Model.Expressions;

namespace ToyLanguageInterpreter.Src.Model.Statements
{
    class Print : IStatement
    {
        private IExpression exp; 

        public Print(IExpression exp)
        {
            this.exp = exp;
        }

        public override string ToString()
        {
            return "print(" + exp.ToString() + ")";
        }

        public ProgramState Exec(ProgramState state)
        {
            state.AppendOutput(exp.Eval(state) + "\n");
            return null;
        }
    }
}
